### ISSUE TITLE
Array[T] of `const` length get parsed as just T type with -f:ast2, if that const length var is not resolved

### ISSUE THREAD

**Issue by kaushalmodi** [nimterop/nimterop#212]
Here's a minimal reproducible example:

### `misc.h`

```h
#ifndef _MISC_H_
#define _MISC_H_

#include <stdint.h>
#include <string>

const uint8_t   SOME_CONST = 8;

#endif // !MISC_H
```

### `test.h`

```h
#ifndef TEST_H
#define TEST_H

#include <vector>

#include "misc.h"
#include <stdio.h>
#include <string.h>

struct parent_struct_s
{
    some_struct_s s[SOME_CONST];
};

#endif // !TEST_H
```

## Wrong output

Running `~/.nimble/pkgs/nimterop-0.5.2/nimterop/toast --preprocess -m:cpp -f:ast2 --recurse test.h --pnim` gives:

Note that `s*` is not an array!: `s*: some_struct_s`

```nim
# Generated at 2020-05-12T12:45:17-04:00
# Command line:
#   /home/kmodi/.nimble/pkgs/nimterop-0.5.2/nimterop/toast --preprocess -m:cpp -f:ast2 --recurse test.h --pnim

{.hint[ConvFromXtoItselfNotNeeded]: off.}

import nimterop/types


{.pragma: imptestHdr, header: "/home/kmodi/sandbox/nim/bug_reports/nimterop_arrays_in_struct/test.h".}
{.experimental: "codeReordering".}
type
  parent_struct_s* {.bycopy, imptestHdr, importc: "struct parent_struct_s".} = object
    s*: some_struct_s
```

## Somewhat correct output (not sure why the const is not exporting to Nim)

Running `~/.nimble/pkgs/nimterop-0.5.2/nimterop/toast --preprocess -m:cpp -f:ast1 --recurse test.h --pnim`:

Note that this is now correct: `s*: array[SOME_CONST, some_struct_s]`

```nim
# Generated at 2020-05-12T12:44:06-04:00
# Command line:
#   /home/kmodi/.nimble/pkgs/nimterop-0.5.2/nimterop/toast --preprocess -m:cpp -f:ast1 --recurse test.h --pnim

{.hint[ConvFromXtoItselfNotNeeded]: off.}

import nimterop/types

const
  headertest {.used.} = "/home/kmodi/sandbox/nim/bug_reports/nimterop_arrays_in_struct/test.h"

{.pragma: imptest, importc, header: headertest.}
{.pragma: imptestC, imptest, cdecl.}

type
  parent_struct_s* {.importc: "struct parent_struct_s", header: headertest, bycopy.} = object
    s*: array[SOME_CONST, some_struct_s]


# Declaration 'SOME_CONST' skipped

{.hint: "The legacy wrapper generation algorithm is deprecated and will be removed in the next release of Nimterop.".}
{.hint: "Refer to CHANGES.md for details on migrating to the new backend.".}
```

**Comment by genotrance** [comment]
Since the const isn't defined, exprparser probably throws out the array size. @jyapayne, any ideas?

**Comment by jyapayne** [comment]
Hmm, there must be an nkNone that needs to be checked after exprparser gets called when parsing the struct definition. exprparser would just return nkNone if the symbol isn't defined, so it's up to the caller to do something with it.

**Comment by jyapayne** [comment]
Probably here: https://github.com/nimterop/nimterop/blob/master/nimterop/ast2.nim#L954. We might want to skip if size is nkNone

**Comment by kaushalmodi** [comment]
I think that earlier if I had 

```
struct parent_struct_s
{
    some_struct_s s[SOME_CONST];
};
```

, the fact that `s` is an array of length `SOME_CONST` still got retained even if SOME_CONST could not be evaluated or found in the scope.

Now, the `SOME_CONST` is replaced with its value if that const is found, else that array is converted to a scalar.

I preferred the older behavior because the user could grep for the `SOME_CONST` to see where it got used in the Nim wrapper (it should be used at the same places as in the original header files). Now the SOME_CONST is see only in the const declaration and everywhere else, SOME_CONST gets expanded to its value.

__*Before*__

Whether or not `SOME_CONST` gets resolved:

```
s*: array[SOME_CONST, some_struct_s]
```

__*After*__

If `SOME_CONST` gets resolved (say, it's value is 8), the const identifier is replaced by toast:

```
s*: array[8, some_struct_s]
```

If `SOME_CONST` does not get resolved, we get this (which I believe is very unexpected):

```
s*: some_struct_s
```

**Comment by genotrance** [comment]
Just merged @jyapayne changes, @kaushalmodi please test when you get a chance.

**Comment by kaushalmodi** [comment]
I confirm that this issue is fixed (as of testing nimterop 0.5.4). Thank you @jyapayne and @genotrance!
